REVIEWING.md

```diff
@@ -0,0 +1,133 @@
+# Code Review in Cluster API
+
+## Goal of this document
+
+- To help newcomers to the project in implementing better PRs given the knowledge of what will be evaluated 
+  during the review.
+- To help contributors in stepping up as a reviewer given a common understanding of what are the most relevant
+  things to be evaluated during the review.
+
+IMPORTANT: improving and maintaining this document is a collaborative effort, so we are encouraging constructive
+feedback and suggestions.
+
+   * [Code Review in Cluster API](#code-review-in-cluster-api)
+      * [Goal of this document](#goal-of-this-document)
+      * [Resources](#resources)
+      * [Definition](#definition)
+      * [CAPI Design considerations](#capi-design-considerations)
+         * [Controller reentrancy](#controller-reentrancy)
+         * [API design](#api-design)
+            * [Serialization](#serialization)
+            * [Owner References](#owner-references)
+         * [The Cluster API contract](#the-cluster-api-contract)
+
+## Resources
+
+- [Contributor Summit NA 2019: Keeping the Bar High - How to be a bad ass Code Reviewer](https://www.youtube.com/watch?v=OZVv7-o8i40)
+- [Code Review Developer Guide - Google](https://google.github.io/eng-practices/review/)
+- [The Gentle Art Of Patch Review](https://sage.thesharps.us/2014/09/01/the-gentle-art-of-patch-review/)
+- [Writing inclusive documentation](https://developers.google.com/style/inclusive-documentation)
+
+## Definition
+
+(from [Code Review Developer Guide - Google](https://google.github.io/eng-practices/review/)) 
+
+A code review is a process where someone other than the author(s) of a piece of code examines that code
+
+Code reviews should look at:
+
+- Design: Is the code well-designed and appropriate for your system?
+- Functionality: Does the code behave as the author likely intended? Is the way the code behaves good for its users?
+- Complexity: Could the code be made simpler? Would another developer be able to easily understand and use this code
+  when they come across it in the future?
+- Tests: Does the code have correct and well-designed automated tests?
+- Naming: Did the developer choose clear names for variables, classes, methods, etc.?
+- Comments: Are the comments clear and useful?
+- Style: Does the code follow our style guides?
```

Comment:
is there a relevant link for "our style guides"?

Reply:
We don't have one (yet), I proposed to remove this section from this document and go in details, rather than providing a summary which should be in the contributing guide

Reply:
removed as per comment above